connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -40,17 +46,16 @@
         void onRequestLocationPermission();
     }
 
+    private static final String WORK_ID_CONNECTION_POLLING = "connection_refresh_polling";
+    private static final long CONNECTION_REFRESH_POLLING_INTERVAL = 1;
+
     private static ConnectLocation INSTANCE = null;
+
     final GeofenceProvider geofenceProvider;
     final ConnectionApiClient connectionApiClient;
-    private final WorkManager workManager;
-
-    private LocationPermissionCallback permissionCallback;
```

Comment:
We might run into a memory leak here if we keep a reference of the callback in this class. I moved it to a different class and set that directly with the ConnectButton.